addons/account/static/src/components/account_payment_field/account_payment.xml

```diff
@@ -92,6 +92,10 @@
                             <td><strong>Journal: </strong></td>
                             <td><t t-out="props.journal_name"/><span t-if="props.payment_method_name"> (<t t-out="props.payment_method_name"/>)</span></td>
                         </tr>
+                        <tr t-if="props.company_name">
+                            <td><strong>Company: </strong></td>
```

Comment:
@tsb-odoo Don't we wanna label that "Branch" rather than Company?

Reply:
It might either be the parent company or a branch, so neither one makes full sense. We could also do `Company/Branch:`? But we'll let @tsb-odoo decide 🙂